packages/amplify_datastore/android/src/test/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePluginTest.kt

```diff
@@ -126,11 +129,128 @@ class AmplifyDataStorePluginTest {
         verify(mockResult, times(1)).success(emptyList<FlutterSerializedModel>())
     }
 
+    @Test
+    fun test_Delete_Success_Result_No_Predicates() {
+
+        var modelData: HashMap<String, Any> = hashMapOf(
+            "id" to "43036c6b-8044-4309-bddc-262b6c686026",
+            "title" to "Title 2",
+            "created" to "2020-02-20T20:20:20-08:00"
+        )
+
+        var instance = SerializedModel.builder()
+                .serializedData(modelData)
+                .id(modelData["id"] as String)
+                .modelName("Post")
+                .build()
+
+        var dataStoreItemChange = DataStoreItemChange.builder<SerializedModel>()
+                .item(instance)
+                .initiator(DataStoreItemChange.Initiator.LOCAL)
+                .itemClass(SerializedModel::class.java)
+                .type(DataStoreItemChange.Type.DELETE)
+                .randomUuid()
+                .build()
+
+        doAnswer { invocation: InvocationOnMock ->
+            (invocation.arguments[2] as Consumer<DataStoreItemChange<SerializedModel>>).accept(
+                    dataStoreItemChange)
+            null as Void?
+        }.`when`(mockAmplifyDataStorePlugin).delete(any(), any(QueryPredicate::class.java),
+                ArgumentMatchers.any<
+                        Consumer<DataStoreItemChange<SerializedModel>>>(),
+                ArgumentMatchers.any<Consumer<DataStoreException>>())
+
+        flutterPlugin.onDelete(mockResult,
+                readMapFromFile("delete_api",
+                        "request/instance_no_predicate.json",
+                        HashMap::class.java) as HashMap<String, Any>)
+
+        verify(mockResult, times(1)).success(
+               FlutterSerializedModel(instance).toMap())
```

Comment:
We are missing the android library's call assertion (that it was called with expected parameters). I've added those insertions in the mock answer, but will be changing it to ArgumentCaptor as Jameson suggested.

Reply:
Mind if I wait and follow your example?